### PR TITLE
Reduce certificate expiry time

### PR DESCRIPTION
**Pull request by joedborg**
Trying to address https://github.com/ubuntu/microk8s/issues/945

### CODE REVIEW

**Comment by joedborg** [comment]
This is still blocked by both Chrome and Safari (although I believe this is a bug in Safari as it lets you add the exception and later acknowledges this), but means that the dates on certificates are now valid.

File: microk8s-resources/actions/common/utils.sh [hunk 1]
```diff
@@ -366,6 +375,26 @@ produce_certs() {
     fi
 }
 
+render_ca_conf() {
+    # Render ca.conf.template to csr.conf.rendered
+
+    local IP_ADDRESSES="$(get_ips)"
+
+    cp ${SNAP_DATA}/certs/ca.conf.template ${SNAP_DATA}/certs/ca.conf.rendered
+    if ! [ "$IP_ADDRESSES" == "127.0.0.1" ] && ! [ "$IP_ADDRESSES" == "none" ]
```
Comment (ktsakalozos) [inline microk8s-resources/actions/common/utils.sh]: This part does not look right. 

The `ca.conf.template` does not have a MOREIPS line. I think we do not need this part.

File: microk8s-resources/actions/common/utils.sh [hunk 2]
```diff
@@ -323,26 +323,35 @@ produce_certs() {
         fi
     done
 
+    render_ca_conf
+    if ! [ -f "${SNAP_DATA}/certs/ca.conf" ]; then
+        echo "changeme" >  "${SNAP_DATA}/certs/ca.conf"
+    fi
+
+    if ! "${SNAP}/usr/bin/cmp" -s "${SNAP_DATA}/certs/ca.conf.rendered" "${SNAP_DATA}/certs/ca.conf"; then
+        cp ${SNAP_DATA}/certs/ca.conf.rendered ${SNAP_DATA}/certs/ca.conf
```
Comment (ktsakalozos) [inline microk8s-resources/actions/common/utils.sh]: This does not look right. I think that if there is a change in the ca we should set the force flag on this function so all related certificates are recreated.

File: microk8s-resources/actions/common/utils.sh [hunk 3]
```diff
@@ -323,26 +323,35 @@ produce_certs() {
...
+    fi
+
     # Generate apiserver CA
     if ! [ -f ${SNAP_DATA}/certs/ca.crt ]; then
-        ${SNAP}/usr/bin/openssl req -x509 -new -nodes -key ${SNAP_DATA}/certs/ca.key -subj "/CN=10.152.183.1" -days 10000 -out ${SNAP_DATA}/certs/ca.crt
+        ${SNAP}/usr/bin/openssl req -x509 -new -sha256 -nodes -key ${SNAP_DATA}/certs/ca.key -subj "/CN=10.152.183.1" -out ${SNAP_DATA}/certs/ca.crt -config ${SNAP_DATA}/certs/ca.conf
```
Comment (ktsakalozos) [inline microk8s-resources/actions/common/utils.sh]: This creates the certificate only the first time. Should we follow the pattern we have for the rest of the certificates where is someone changes the template we trigger the re-issue of the certs?

File: microk8s-resources/actions/dashboard.yaml [hunk 4]
```diff
@@ -167,18 +167,14 @@ spec:
     spec:
       containers:
         - name: kubernetes-dashboard
-          image: kubernetesui/dashboard:v2.0.0-beta5
+          image: kubernetesui/dashboard:v2.0.0-rc5
```
Comment (ktsakalozos) [inline microk8s-resources/actions/dashboard.yaml]: We may want to have this as a separete PR so it does not get blocked by the rest of the work we are doing here.

**Comment by balchua** [comment]
Safari is reducing the certificate validity to 1 year.  Im not a mac user, but just wonderin if that will have an impact on this PR.

**Comment by ktsakalozos** [comment]
> Safari is reducing the certificate validity to 1 year

@balchua, @joedborg, I see bot of you agree on a 365 certificate validity so lets go with that (reverted the last commit).

**Pull request merged**